packages/relay/src/lib/eth.ts

```diff
@@ -2531,11 +2544,12 @@ export class EthImpl implements Eth {
     if (blockResponse == null) return null;
     const timestampRange = blockResponse.timestamp;
     const timestampRangeParams = [`gte:${timestampRange.from}`, `lte:${timestampRange.to}`];
-    const contractResults = await this.mirrorNodeClient.getContractResults(
+    const contractResults = await this.mirrorNodeClient.getContractResultWithRetry(
```

Comment:
Are we just being cautious or does this case actually need to have retry support?

Reply:
The `contractResults` here was a contributing factor to the 500 errors. In the rare cases mentioned, `contractResults` contains null fields. When passed to `buildReceiptRootHashes`, the function does not handle these null values correctly, resulting in 500 errors. Therefore, this scenario requires a retry if such a rare case occurs.

Reply:
Since these cases occur only on the mainnet and at the time of the transaction, I plan to release these updates that include additional debugging information for improved analysis. If the issues persist, it will be easier to pinpoint the root causes and escalate them to the mainnet team more effectively.